recipes/qt/5.x.x/conanfile.py

```diff
@@ -0,0 +1,710 @@
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            del self.options.with_icu
+            del self.options.with_fontconfig
+            del self.options.with_libalsa
+        if self.settings.compiler == "apple-clang":
+            if tools.Version(self.settings.compiler.version) < "10.0":
+                raise ConanInvalidConfiguration("Old versions of apple sdk are not supported by Qt (QTBUG-76777)")
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "5.3":
+            del self.options.with_mysql
+        if self.settings.os == "Windows":
+            self.options.with_mysql = False
+            self.options.opengl = "dynamic"
+        if self.settings.os == "Macos":
+            del self.settings.os.version
+
+    def configure(self):
+        #if self.settings.os != 'Linux':
+        #         self.options.with_libiconv = False # QTBUG-84708
+
+        if self.options.widgets and not self.options.gui:
+            raise ConanInvalidConfiguration("using option qt:widgets without option qt:gui is not possible. "
+                                            "You can either disable qt:widgets or enable qt:gui")
+        if not self.options.gui:
+            del self.options.opengl
+            del self.options.with_vulkan
+            del self.options.with_freetype
+            del self.options.with_fontconfig
+            del self.options.with_harfbuzz
+            del self.options.with_libjpeg
+            del self.options.with_libpng
+
+        if not self.options.qtmultimedia:
+            del self.options.with_libalsa
+            del self.options.with_openal
+
+        if self.options.qtwebengine:
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("Static builds of Qt Webengine are not supported")
+
+            if tools.cross_building(self.settings, skip_x64_x86=True):
+                raise ConanInvalidConfiguration("Cross compiling Qt WebEngine is not supported")
+
+            if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
+
+        if self.settings.os == "Android" and self.options.get_safe("opengl", "no") == "desktop":
+            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+
+        if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") == "dynamic":
+            raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
+
+        if self.options.get_safe("with_fontconfig", False) and not self.options.get_safe("with_freetype", False):
+            raise ConanInvalidConfiguration("with_fontconfig cannot be enabled if with_freetype is disabled.")
+
+        if self.options.multiconfiguration:
+            del self.settings.build_type
```

Comment:
Just a sanity check, does this option make sense? (no experience with Qt)

Reply:
To play devil's advocate, when building my application I would use either debug or release of QT but never both simultaneously? What advantage is there in having a package with both?

Reply:
according to conan, it's needed for visual studio and xcode: https://docs.conan.io/en/latest/integrations/build_system/cmake/cmake_multi_generator.html?highlight=cmake_multi